sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -266,7 +266,7 @@ case class Union(
         firstAttr.withNullability(nullable)
       } else {
         AttributeReference(firstAttr.name, newDt, nullable, firstAttr.metadata)(
-          firstAttr.exprId, firstAttr.qualifier)
+          NamedExpression.newExprId, firstAttr.qualifier)
```

Comment:
NOTE: Found by the integrity check: https://github.com/apache/spark/pull/29585/files#diff-27c76f96a7b2733ecfd6f46a1716e153R224

Reply:
Is this the only place found by the check https://github.com/apache/spark/pull/29585/files#diff-27c76f96a7b2733ecfd6f46a1716e153R224 ?

Reply:
And the bug in `StreamSuite`.

Reply:
This change is risky. We are generating new expr ID in `def output`, which means the output of this plan is changing everytime you call `output`.

Reply:
maybe `Union` should have a `val output` in the constructor, like `Generate`.

Reply:
okay, I'll check it around.